modules/gltf/gltf_document.cpp

```diff
@@ -3196,10 +3204,60 @@ Error GLTFDocument::_parse_images(Ref<GLTFState> state, const String &p_base_pat
 		// Now we've done our best, fix your scenes.
 		if (img.is_null()) {
 			ERR_PRINT(vformat("glTF: Couldn't load image index '%d' with its given mimetype: %s.", i, mimetype));
+
 			state->images.push_back(Ref<Texture2D>());
+			state->source_images.push_back(Ref<Image>());
 			continue;
 		}
-		state->images.push_back(ImageTexture::create_from_image(img));
+		if (state->handle_binary_image == GLTFState::GLTF_DOCUMENT_DISCARD_TEXTURES) {
+			state->images.push_back(Ref<Texture2D>());
+			state->source_images.push_back(Ref<Image>());
+			continue;
+		} else if (state->handle_binary_image == GLTFState::GLTF_DOCUMENT_EXTRACT_TEXTURES) {
+			String extracted_image_name = image_name.get_file().get_basename();
+			img->set_name(extracted_image_name);
+			if (state->base_path.is_empty()) {
+				state->images.push_back(Ref<Texture2D>());
+				state->source_images.push_back(Ref<Image>());
+			} else if (img->get_name().is_empty()) {
+				WARN_PRINT(vformat("glTF: Image index '%d' couldn't be named. Skipping it.", i));
+				state->images.push_back(Ref<Texture2D>());
+				state->source_images.push_back(Ref<Image>());
+			} else {
+				String file_path = state->get_base_path() + state->filename.get_basename() + "_" + img->get_name() + ".png";
```

Comment:
sanitize

Reply:
Can you provide example code?

Reply:
does it also need de-duplication (accounting for non-case-sensitive filesystems) or handling of images which don't have names?

Reply:
If the buffer view is nameless, it is set to the number.

If the buffer has duplicates.. I don't think that's allowed in gltf.

Reply:
The issue is they may have duplicates after accounting for case-insensitivity in filesystems. This could lead to behavior which breaks only on windows.

Reply:
Of course, now we start getting into other filesystem quirks like unicode normalization such as on APFS on macos: https://twitter.com/Foone/status/1369516978029793283

Honestly, for our case, I'm not as concerned about collisions (duplicate names or case sensitivity or normalization). So let's shelve this whole part of the concern.

Let's only fix cases that are outright illegal (such as double-quotes or pipe characters), which is what that `_make_extname` function does.

Reply:
Need to check what happens when an illegal name is saved on windows and is renamed.

Reply:
Added a local solution rather than a global solution to Saver.